runtime/src/accounts_db.rs

```diff
@@ -977,11 +994,17 @@ impl AccountsDB {
 
     pub fn process_stale_slot(&self) -> usize {
         let mut measure = Measure::start("stale_slot_shrink-ms");
-        let count = if let Some(slot) = self.next_shrink_slot() {
```

Comment:
`next_shrink_slot` is artifically moved inside `shrink_stale_slot` by passing `candidates` to indicate locked `candidates`'s intended lifetime as a code.